### PR TITLE
Trilinos: New ATS2 environment is modifying a system level ENV variable that breaks Spectrum

### LINKED ISSUE #6875: Trilinos: New ATS2 environment is modifying a system level ENV variable that breaks Spectrum
## Bug Report
@trilinos/framework (I think?)

### Description
In the ATS2 `environment.sh` the `TMPDIR` variable is modified. This causes problems with Spectrum (and perhaps other system level tools) that depend on `TMPDIR`
```
atdm/ats2/environment.sh:174:  export TMPDIR=/tmp/$(whoami)
atdm/ats2/environment.sh:175:  if [[ ! -d ${TMPDIR} ]] ; then
atdm/ats2/environment.sh:176:    echo "Creating ${TMPDIR} for nvcc wrapper!"
atdm/ats2/environment.sh:177:    mkdir ${TMPDIR}
```

### Steps to Reproduce
1. load the ats2 environment on Vortex
2. jsrun will fail after

@e10harvey @bartlettroscoe 

I will be submitting a PR that fixes this

### PR DESCRIPTION
**Pull request by jjellio**
The ATS2 ENV is changing TMPDIR, which is usually a reserved ENV.

This causes a problem with jsrun RM connections.

This patch renames that variable and prefixes it with Trilinos

It isn't obvious this variable is used by the framework...
If it is used, then where it is used needs to be updated as well.

Fixes #6875

## Testing
Applying this patch fixed a jsrun issue on Vortex

### CODE REVIEW

File: cmake/std/atdm/ats2/environment.sh [hunk 1]
```diff
@@ -171,10 +171,10 @@ if [[ "$ATDM_CONFIG_COMPILER" == "CUDA-10.1.243_"* ]]; then
...
   export CUDA_LAUNCH_BLOCKING=1
   export CUDA_MANAGED_FORCE_DEVICE_ALLOC=1
-  export TMPDIR=/tmp/$(whoami)
-  if [[ ! -d ${TMPDIR} ]] ; then
-    echo "Creating ${TMPDIR} for nvcc wrapper!"
-    mkdir ${TMPDIR}
+  export TRILINOS_NVCC_WRAPPER_TMPDIR=/tmp/$(whoami)
+  if [[ ! -d ${TRILINOS_NVCC_WRAPPER_TMPDIR} ]] ; then
+    echo "Creating ${TRILINOS_NVCC_WRAPPER_TMPDIR} for nvcc wrapper!"
+    mkdir ${TRILINOS_NVCC_WRAPPER_TMPDIR}
```
Comment (bartlettroscoe) [inline cmake/std/atdm/ats2/environment.sh]: @jjellio and @e10harvey, the only reason this code was added was to get around nvcc_wrapper build errors when that directory did not exist on a given node.  Is there a reason why we should even be creating this env var `TRILINOS_NVCC_WRAPPER_TMPDIR`?  What is it used for?  If it is used for nothing, we should just delete it.

**Comment by bartlettroscoe** [comment]
I can't currently test this as I can't log into my SNL account :-(

File: cmake/std/atdm/ats2/environment.sh (hunk 1 above)
Comment (e10harvey) [reply cmake/std/atdm/ats2/environment.sh]: `TRILINOS_NVCC_WRAPPER_TMPDIR` does not need to exported.

Comment (bartlettroscoe) [reply cmake/std/atdm/ats2/environment.sh]: @jjellio and @e10harvey, then can this PR just delete the `TRILINOS_NVCC_WRAPPER_TMPDIR` and `TMPDIR` vars being set set in this script all together?

**Comment by bartlettroscoe** [comment]
@jjellio and @e10harvey, fyi, I just pushed the commit https://github.com/trilinos/Trilinos/pull/6876/commits/0075f3d8f27282cb651141e6fca5cba617dd2ccb.  I am testing it now on 'vortex'.

File: cmake/std/atdm/ats2/environment.sh [hunk 2]
```diff
@@ -171,10 +171,9 @@ if [[ "$ATDM_CONFIG_COMPILER" == "CUDA-10.1.243_"* ]]; then
   # CUDA Settings
   export CUDA_LAUNCH_BLOCKING=1
   export CUDA_MANAGED_FORCE_DEVICE_ALLOC=1
-  export TRILINOS_NVCC_WRAPPER_TMPDIR=/tmp/$(whoami)
-  if [[ ! -d ${TRILINOS_NVCC_WRAPPER_TMPDIR} ]] ; then
-    echo "Creating ${TRILINOS_NVCC_WRAPPER_TMPDIR} for nvcc wrapper!"
-    mkdir ${TRILINOS_NVCC_WRAPPER_TMPDIR}
+  if [[ ! -d /tmp/${USER} ]] ; then
+    echo "Creating /tmp/${USER} for nvcc wrapper!"
```
Comment (e10harvey) [inline cmake/std/atdm/ats2/environment.sh]: If this is for nvcc_wrapper and nvcc_wrapper is using `"${TMPDIR:-/tmp}"/$(whoami)`, I think we need to set `"${TMPDIR:-/tmp}"/$(whoami)`. I am testing this from https://github.com/e10harvey/Trilinos/tree/topic/jsrun_bug_swat on vortex now....

**Comment by bartlettroscoe** [comment]
FYI: I tested commit ??? by running the following on 'vortex':

```
$ env Trilinos_PACKAGES=Kokkos,Teuchos,Tpetra \
  ./ctest-s-local-test-driver.sh ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt

***
*** ./ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/home/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'vortex60' matches known ATDM host 'vortex' and system 'ats2'
Setting compiler and build options for build-name 'default'
Using ats2 compiler stack GNU-7.3.1_SPMPI-2019.06.24 to build DEBUG code with Kokkos node type SERIAL

Lmod is automatically replacing "xl/2019.12.23" with "gcc/7.3.1".


Due to MODULEPATH changes, the following have been reloaded:
  1) spectrum-mpi/rolling-release


The following have been reloaded with a version change:
  1) spectrum-mpi/rolling-release => spectrum-mpi/2019.06.24


Running builds:
    ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt

Fri Feb 21 13:47:03 MST 2020

Running Jenkins driver Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt.sh ...

    See log file Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt/smart-jenkins-driver.out

real    58m29.908s
user    0m2.126s
sys     0m0.699s

99% tests passed, 2 tests failed out of 401
99% tests passed, 2 tests failed out of 401

Fri Feb 21 14:45:33 MST 2020

Done running all of the builds!
```

which posted to CDash:

* [platformTrilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt-exp](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&parentid=5253640) (pass=299, fail=2)
* [https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&parentid=5253644](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&parentid=5253644) (pass=399, fail=2)

The 2 failing tests are Tpetra tests we already know about from #6796 and #6883.

**Comment by bartlettroscoe** [comment]
@bathmatt, I pushed commits d68e62a629 and 5c5c6f8ee5 to this PR that should allow you to use the installed trilinos_jsrun just fine (and this PR should also avoid other problems with using jsrun). 

I tested a local install  on 'vortex' and saw:

```
$ ls -l install/share/atdm-trilinos/ats2/
total 66
-rw-rw-r-- 1 rabartl rabartl  444 Jan 31 09:01 all_supported_builds.sh
-rw------- 1 rabartl rabartl 2906 Feb 18 12:37 custom_builds.sh
-rw-rw-r-- 1 rabartl rabartl 9955 Feb 21 11:43 environment.sh
-rwxr-xr-x 1 rabartl rabartl 5470 Feb 18 12:37 trilinos_jsrun
-rwxr-xr-x 1 rabartl rabartl 1857 Feb 18 12:37 trilinos_jsrun_test_me.sh
drwxrwxr-x 2 rabartl rabartl 4096 Feb 21 15:34 tweaks
```

Can you please pull the branch and merge it into your local Trilinos branch and see if it works for you?

Otherwise, the PR tester is broken until at least Monday so this will not be getting on to 'develop' until then.

**Comment by bartlettroscoe** [comment]
FYI: With the Trilinos PR testing system down with no ETA to fix, I manually merged this branch to the 'atdm-nightly-manual-updates' branch just now in commit 47b673b so it will be in the 'atdm-nightly' branch tonight and we will see this running tomorrow.

**Comment by bartlettroscoe** [comment]
FYI: It seems the name [TMPDIR](https://en.wikipedia.org/wiki/TMPDIR) is a reserved Unix/POSIX system var.  However, it seems there is some precedent for users being told to sent this var:

* https://www.ibm.com/support/knowledgecenter/SSTRGZ_11.4.0/com.ibm.swg.im.iis.repl.qrepl.doc/topics/iiyrqenvtempdir.html

**Pull request merged**